conformance/lang/expressions/regexp/regexp_semantics.balt

```diff
@@ -0,0 +1,719 @@
+};
+
+function init() {
+    ReRec1 r1 = {};
+    io:println(r1.re1 is string:RegExp); // @output true
+    io:println(r1.re1 == re `ABC`); // @output true
+}
+
+Test-Case: output
+Description: Test regular expression as table field.
+Labels: RegExp, table-constructor-expr, table-type-descriptor
+
+type Foo record {
+    readonly int id;
+    string:RegExp re;
+};
+
+table<Foo> key(id) t = table [
+    {id : 1, re : re `ABC`},
+    {id : 2, re : re `DEF`}
+];
+
+function init() {
+    io:println(t.get(1).re is string:RegExp); // @output true
+    io:println(t.get(1).re == re `ABC`); // @output true
+
+    io:println(t.get(2).re is string:RegExp); // @outout true
+    io:println(t.get(2).re == re `DEF`); // @output true
+}
+
+Test-Case: output
+Description: Test regular expression as table field.
+Labels: RegExp, table-constructor-expr, table-type-descriptor
+
+type Foo record {
+    readonly int id;
+    string:RegExp re;
+};
+
+table<Foo> key(id) t = table [
+    {id : 1, re : re `ABC`},
+    {id : 2, re : re `DEF`}
+];
+
+function init() {
+    io:println(t.get(1).re is string:RegExp); // @output true
+    io:println(t.get(1).re == re `ABC`); // @output true
+
+    io:println(t.get(2).re is string:RegExp); // @outout true
+    io:println(t.get(2).re == re `DEF`); // @output true
+}
+
+Test-Case: output
+Fail-Issue: https://github.com/ballerina-platform/ballerina-lang/issues/38679 
+Description: Test 1 for regular expression as table key.
+Labels: RegExp, table-constructor-expr, table-type-descriptor
+
+type Foo record {
+    readonly int id;
+    readonly string:RegExp re;
+};
+
+table<Foo> key(re) t = table [
+    {id : 1, re : re `ABC`},
+    {id : 2, re : re `DEF`}
+];
+
+function init() {
+    io:println(t.get(re `ABC`).re is string:RegExp); // @output true
+    io:println(t.get(re `ABC`).re == re `ABC`); // @output true
+
+    io:println(t.get(re `DEF`).re is string:RegExp); // @output true
+    io:println(t.get(re `DEF`).re == re `DEF`); // @output true
+}
+
+Test-Case: output
+Fail-Issue: https://github.com/ballerina-platform/ballerina-lang/issues/38679
+Description: Test 2 for regular expression as table key.
+Labels: RegExp, table-constructor-expr, table-type-descriptor
+
+type Foo record {
+    readonly int id;
+    readonly string:RegExp re;
+};
+
+function init() {
+    table<Foo> key(re) t = table [
+        {id: 1, re: re `ABC`},
+        {id: 2, re: re `DEF`}
+    ];
+
+    io:println(t.get(re `ABC`).re is string:RegExp); // @output true
+    io:println(t.get(re `ABC`).re == re `ABC`); // @output true
+
+    io:println(t.get(re `DEF`).re is string:RegExp); // @output true
+    io:println(t.get(re `DEF`).re == re `DEF`); // @output true
+}
+
+Test-Case: output
+Description: Test 1 for regular expression as list member.
+Labels: RegExp, list-constructor-expr
+
+function init() {
+    string:RegExp[] regArr = [re `ABC`, re `([A-D]+|[H-Z]*)|C{2,9}`, re `[a-z]`];
+    io:println(regArr.length().toString()); // @output 3
```

Comment:
do  io:println(regArr.length() is 3) instead

Reply:
There are a few other places which have the same check. Let's fix them too.